(WIP) LAMMPS Molecular Dynamics enhancements

badarsh2 opened the pull request:
This PR comprises of commits that are aimed at adding LAMMPS MD trajectory handling compatibility to Avogadro. 
To-do list:

- Workaround for atom types being considered as atomic numbers
- Affine transformation capabilities for unit cell (origin shift)
- Addition of support for other auxiliary field types in "Atoms" item

ghutchis commented on avogadro/io/lammpsformat.cpp:
```diff
@@ -0,0 +1,368 @@
+
+  This source file is part of the Avogadro project.
+
+  Copyright 2013 Kitware, Inc.
```
It's 2018...

ghutchis commented on avogadro/io/lammpsformat.cpp:
```diff
@@ -0,0 +1,368 @@
+  }
+
+  // This format has no connectivity information, so perceive basics at least.
+  mol.perceiveBondsSimple();
```
Should this be optional? Consider if I read in a simulation of a gold monolayer, I probably don't want bonds.

I'll be adding in actual commands in the Build menu to allow users to perceive bonds after import, so I'd drop this for now.

ghutchis commented:
It's probably best to split some of these "to do" items into separate pull requests. I haven't looked at the Travis failure, but IMHO this is a good amount for a first PR (i.e., this is a self-contained piece).

cryos commented:
I haven't had a chance to look through the code, but I absolutely agree with @ghutchis that we want to keep pull requests reasonably small and self-contained. We don't need feature complete in one pull request, but can work towards it. Smaller changes let us look more carefully at the impact of what is being changed.

cryos commented:
Please squash 6d2061c into 469f158, the CMakeLists change will just make history harder to follow. I can help you out with editing history if you have trouble, but [this](https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d) is a good introduction.

badarsh2 commented:
Sure. From now on, I shall keep pull requests small. I've removed the to-do list for now and shall work on those items in future PRs.
I've squashed the two commits and pushed new commits to implement the suggested changes (hope I've done it right). With regard to the travis failure, I'm not able to figure out what is going wrong. I've run clang-format for the new files.

cryos commented:
If you click through clang-format shows you what it sees, and how it would change that to match our format in the diff. I guess when you run clang-format you are not getting the same output? That is odd as we just tightened the format file to work across more versions - what version are you running, and what command line are you using?

badarsh2 commented:
I gues I've fixed it now... I was running the wrong clang-format command all along. Sorry about that :(

cryos commented on avogadro/io/lammpsformat.cpp:
```diff
@@ -0,0 +1,413 @@
+
+    unsigned char atomicNum(0);
+    atomicNum =
+      static_cast<unsigned char>(lexicalCast<short int>(tokens[type_idx - 2]));
```
I don't understand the lexicalCast *and* the static_cast - did the lexicalCast<unsigned char> fail?

cryos commented on avogadro/io/lammpsformat.cpp:
```diff
@@ -0,0 +1,413 @@
+    // scale_x = 0. if coordinates are cartesian and 1 if fractional (scaled)
+    getline(inStream, buffer);
+    labels = vector<string>(split(buffer, ' '));
+    for (size_t i = 0; i < labels.size(); i++) {
```
Prefer pre-increment unless post-increment is needed.

cryos commented on avogadro/io/lammpsformat.cpp:
```diff
@@ -0,0 +1,413 @@
+                                 Vector3(tilt_xy, y_max - y_min, 0),
+                                 Vector3(tilt_xz, tilt_yz, z_max - z_min)));
+
+    positions.clear();
```
Is this needed? It looks to me like it is about to go out of scope, and so would be cleared, and then a new one created on the next loop.

badarsh2 commented:
I've fixed the issues raised above. I shall work on the tests soon.

cryos commented:
Pushed a tiny fix to CMake code to trigger a new CI build, if you remove the (WIP) from the title I think we can merge this (assuming build results look good too).

badarsh2 commented:
Shall I add a test in this branch now itself, or post merging of this PR?

cryos commented:
This PR is merged, you can start a new branch, or simply extend this one. If you extend this one, you will end up creating a new pull request anyway - it doesn't matter too much where you start the branch from in this case, if there were bigger changes in master you would branch from master to reduce the chances of seeing a merge conflict.

badarsh2 commented:
This PR still seems to be unmerged....?

cryos commented:
My mistake, you have quite a few PRs, and the merged icon above fooled me :-) Merged now.
